server/channels/api4/team.go

```diff
@@ -151,7 +151,16 @@ func getTeam(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if (!team.AllowOpenInvite || team.Type != model.TeamOpen) && !c.App.SessionHasPermissionToTeam(*c.AppContext.Session(), team.Id, model.PermissionViewTeam) {
+	isPublicTeam := team.AllowOpenInvite && team.Type == model.TeamOpen
```

Comment:
This is much cleaner, thank you. I had to write out a truth table to double check the first error condition below is the same as the old one, which proves the old way needed to be simplified. :)